server/routerlicious/packages/services-core/src/mongoDocumentRepository.ts

```diff
@@ -0,0 +1,44 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { FindAndOperateDocumentMode, ICollection, IDb, IDocumentRepository } from "./database";
+import { IDocument } from "./document";
+
+export class MongoDocumentRepository implements IDocumentRepository {
+    constructor(private readonly collection: ICollection<IDocument>) {
+    }
+
+    public static create(
+        database: IDb,
+        collectionName: string,
```

Comment:
Why not just pass in ICollection<IDocument>?

Reply:
I wan to hide the `ICollection` as much as possible